### PR TITLE
improve zsh completion

### PR DESCRIPTION
**Pull request by e-kwsm**
* fix error
* remove options that are no longer supported
* add missing options

### CODE REVIEW

File: shell-completion/zsh/_systemd-delta [hunk 1]
```diff
@@ -9,8 +9,8 @@ _systemd-delta_types() {
 }
 
 _arguments \
-    {-h,--help}'[Show this help]' \
-    '--version[Show package version]' \
+    '(- :)'{-h,--help}'[Show this help]' \
+    '(- :)--version[Show package version]' \
```
Comment (keszybz) [inline shell-completion/zsh/_systemd-delta:13]: Hmm, what does this syntax do?

File: shell-completion/zsh/_systemd-inhibit [hunk 2]
```diff
@@ -26,8 +26,10 @@ _systemd-inhibit_what() {
 }
 
 _arguments \
-    {-h,--help}'[Show this help]' \
-    '--version[Show package version]' \
+    '(- *)'{-h,--help}'[Show this help]' \
+    '(- *)--version[Show package version]' \
```
Comment (keszybz) [inline shell-completion/zsh/_systemd-inhibit:30]: And this one??

**Review by keszybz — APPROVED** [review]
I did some testing and it seems to all work great. My only request is for the commit message.

The zsh syntax is strange… Please explain very briefly in the commit message what the syntax with `(…){…}` does. And in particular, why sometimes it's with `*` and `:`.  I'm sure it's documented *somewhere*, but it's not easy to search for, but if you describe it in the commit message, then the next person looking at this can figure it out more easily.

Also, in the commit message, don't say that "support was removed". It's still supported, we just don't want to advertise it.

File: shell-completion/zsh/_systemd-delta (hunk 1 above)
Comment (rpigott) [reply shell-completion/zsh/_systemd-delta:13]: I agree it's worthwhile to carefully explain all the zsh completions in the commit message, but for the record, it is documented in zshcompsys(1) in the section about optspecs in _arguments:

> Each of the forms above may be preceded by a list in parentheses of option names  and  argument  num‐
> bers.  If the given option is on the command line, the options and arguments indicated in parentheses
> will not be offered.  For example, ‘(-two -three 1)-one:...' completes the option ‘-one'; if this ap‐
> pears  on  the  command line, the options -two and -three and the first ordinary argument will not be
> completed after it.  ‘(-foo):...' specifies an ordinary argument completion; -foo will  not  be  com‐
> pleted if that argument is already present.
> 
> Other  items  may  appear in the list of excluded options to indicate various other items that should
> not be applied when the current specification is matched: a single star (\*) for  the  rest  arguments
> (i.e. a specification of the form ‘\*:...'); a colon (:) for all normal (non-option-) arguments; and a
> hyphen (-) for all options.  For example, if ‘(\*)' appears before an option and the option appears on
> the  command  line,  the  list  of remaining arguments (those shown in the above table beginning with
> ‘\*:') will not be completed.

Even this description is a little opaque if you aren't already familiar with zshcompsys. The commit message should explain the intended effect of the change, which is to remove irrelevant completion matches from the completion.

Comment (e-kwsm) [reply shell-completion/zsh/_systemd-delta:13]: As quoted, `(- : )` prevents further completion.

File: shell-completion/zsh/_systemd-inhibit (hunk 2 above)
Comment (e-kwsm) [reply shell-completion/zsh/_systemd-inhibit:30]: Same here.

**Review by keszybz — APPROVED** [review]

**Pull request merged**